x/participation/simulation/simulation.go

```diff
@@ -0,0 +1,248 @@
+	) (simtypes.OperationMsg, []simtypes.FutureOperation, error) {
+		msg := &types.MsgParticipate{}
+		auction, found := RandomAuction(ctx, r, fk)
+		if !found {
+			return simtypes.NoOpMsg(types.ModuleName, msg.Type(), "no valid auction found"), nil, nil
+		}
+
+		tierList := k.ParticipationTierList(ctx)
+		numTiers := len(tierList)
+		if numTiers == 0 {
+			return simtypes.NoOpMsg(types.ModuleName, msg.Type(), "no valid tiers"), nil, nil
+		}
+
+		tierID := RandomTierFromList(r, tierList)
+		tier, _ := types.GetTierFromID(tierList, tierID)
+		simAccount, _, found := RandomAccWithAvailableAllocations(ctx, r, k, accs, tier.RequiredAllocations, auction.GetId())
+		if !found {
+			return simtypes.NoOpMsg(types.ModuleName, msg.Type(), "no account with allocations"), nil, nil
+
+		}
+
+		msg = sample.MsgParticipate(
+			simAccount.Address.String(),
+			auction.GetId(),
+			tierID,
+		)
+
+		txCtx := simulation.OperationInput{
+			R:               r,
+			App:             app,
+			TxGen:           simappparams.MakeTestEncodingConfig().TxConfig,
+			Cdc:             nil,
+			Msg:             msg,
+			MsgType:         msg.Type(),
+			Context:         ctx,
+			SimAccount:      simAccount,
+			AccountKeeper:   ak,
+			Bankkeeper:      bk,
+			ModuleName:      fundraisingtypes.ModuleName,
+			CoinsSpentInMsg: sdk.NewCoins(),
+		}
+		return simulation.GenAndDeliverTxWithRandFees(txCtx)
+	}
+}
+
+func SimulateCreateAuction(
+	ak authkeeper.AccountKeeper,
+	bk bankkeeper.Keeper,
+	fk fundraisingkeeper.Keeper,
+	_ keeper.Keeper,
+) simtypes.Operation {
+	return func(r *rand.Rand, app *baseapp.BaseApp, ctx sdk.Context, accs []simtypes.Account, chainID string,
+	) (simtypes.OperationMsg, []simtypes.FutureOperation, error) {
+		// fundraising simulation params must be set
+		// since the module is not included in the simulation manager
+		params := fundraisingtypes.DefaultParams()
+		fk.SetParams(ctx, params)
+		fee := params.AuctionCreationFee
+		sellCoin := sample.Coin()
+
+		// choose custom fee that only uses the default bond denom
+		// otherwise the custom sellingCoin denom could be chosen
+		customFee, err := simtypes.RandomFees(r, ctx, fee)
+		if err != nil {
+			if err != nil {
+				return simtypes.OperationMsg{},
+					nil,
+					err
+			}
+		}
+
+		simAccount, _, found := RandomAccWithBalance(ctx, r, bk, accs, fee.Add(customFee...))
+		if !found {
+			return simtypes.NoOpMsg(
+					types.ModuleName,
+					fundraisingtypes.MsgCreateFixedPriceAuction{}.Type(),
+					"no account with balance found"),
+				nil,
+				nil
+		}
+
+		startTime := ctx.BlockTime().Add(time.Hour * 24)
+		endTime := startTime.Add(time.Hour * 24 * 7)
+		msg := sample.MsgCreateFixedAuction(simAccount.Address.String(), sellCoin, startTime, endTime)
+
+		mintAmt := sdk.NewCoins(msg.SellingCoin)
+		// must mint and send new coins to auctioneer
+		err = bk.MintCoins(ctx, minttypes.ModuleName, mintAmt)
```

Comment:
We should not directly access keepers in simulation testing. We should use balance set initially for selling coins

Reply:
What do you mean by this?

How will the seller have an initial state with this coin if it is some new denom?

Reply:
> What do you mean by this?

We should only interact with State Transition to actually simulate operations like in a real environment

Accounts have a balance with one denom in the initial state. Isn't it possible to add other denom in the initial state? What I mean is that the balances should be defined before any operation instead of minting them on the fly during the operations

Reply:
I wasn't sure if that was something we could do.  It might be configurable before the sim is run.  I'll look into it

Reply:
Ok, the solution will be to come up  with a custom version of the `AppStateFn` that gives some accounts balances of the staking token as well as some of these custom tokens. 

Then the simulation logic will need to be modified to look for accounts with existing amounts of custom tokens sufficient to start auctions.

Thoughts?

Here is the [function](https://github.com/cosmos/cosmos-sdk/blob/c91f3b26512cc6723ebbbbe986d2842e9c74e698/simapp/state.go#L28).

Reply:
Seems like the right way to go.
We could use a specific denom used as the selling coin for all auctions created